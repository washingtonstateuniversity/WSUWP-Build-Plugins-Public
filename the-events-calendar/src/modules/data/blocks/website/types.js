/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/WEBSITE_BLOCK/SET_INITIAL_STATE`;

export const SET_WEBSITE_URL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_URL`;

export const SET_WEBSITE_LABEL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_LABEL`;
