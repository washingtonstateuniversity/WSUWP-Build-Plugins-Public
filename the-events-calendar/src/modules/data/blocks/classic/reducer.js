/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
