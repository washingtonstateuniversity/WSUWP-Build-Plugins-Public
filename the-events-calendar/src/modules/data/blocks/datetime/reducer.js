/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
