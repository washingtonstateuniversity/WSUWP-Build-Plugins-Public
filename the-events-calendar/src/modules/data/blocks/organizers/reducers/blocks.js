/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
