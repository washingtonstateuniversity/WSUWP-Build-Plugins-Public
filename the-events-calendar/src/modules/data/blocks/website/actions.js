/**
 * Internal dependencies
 */
import * as types from './types';

export const setWebsite = ( url ) => ( {
	type: types.SET_WEBSITE_URL,
	payload: {
		url,
	},
} );

export const setLabel = ( label ) => ( {
	type: types.SET_WEBSITE_LABEL,
	payload: {
		label,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );
