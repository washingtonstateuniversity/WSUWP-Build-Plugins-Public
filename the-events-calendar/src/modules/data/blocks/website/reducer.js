/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	url: undefined,
	label: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_WEBSITE_URL:
			return {
				...state,
				url: action.payload.url,
			};
		case types.SET_WEBSITE_LABEL:
			return {
				...state,
				label: action.payload.label,
			};
		default:
			return state;
	}
};
