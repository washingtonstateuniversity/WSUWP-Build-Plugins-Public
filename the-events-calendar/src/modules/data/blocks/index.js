/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
