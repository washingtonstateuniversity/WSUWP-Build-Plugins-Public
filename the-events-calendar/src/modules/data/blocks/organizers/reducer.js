/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, blocks } from './reducers';

export default combineReducers( {
	blocks,
	classic,
} );
