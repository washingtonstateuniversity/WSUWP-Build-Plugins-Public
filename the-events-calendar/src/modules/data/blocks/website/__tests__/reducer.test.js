/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/website';
import reducer, { DEFAULT_STATE } from '@moderntribe/events/data/blocks/website/reducer';

describe( '[STORE] - Website reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set the website value', () => {
		expect( reducer( DEFAULT_STATE, actions.setWebsite( 'https://tri.be/' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the website label', () => {
		expect( reducer( DEFAULT_STATE, actions.setLabel( 'Modern Tribe' ) ) ).toMatchSnapshot();
	} );
} );
