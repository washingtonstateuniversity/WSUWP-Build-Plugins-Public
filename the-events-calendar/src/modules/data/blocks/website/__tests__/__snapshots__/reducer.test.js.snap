// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Website reducer Should set the website label 1`] = `
Object {
  "label": "Modern Tribe",
  "url": undefined,
}
`;

exports[`[STORE] - Website reducer Should set the website value 1`] = `
Object {
  "label": "",
  "url": "https://tri.be/",
}
`;
